### ISSUE TITLE
Codebuild: Ability to upload coverage and test reports

### ISSUE THREAD

**Issue by blampe** [aws/aws-cli#5686]
**Is your feature request related to a problem? Please describe.**

`aws codebuild` currently only provides read-only access to most resources, more specifically there is no way to append new reports to an existing `ReportGroup`. This prevents me from manually uploading new coverage reports.

This functionality would be helpful when running tests outside of the Codebuild environment, for example on Github. The `aws-cli` project itself uploads coverage reports to codecov.io, but not to a code coverage report in Codebuild. 

**Describe the solution you'd like**

Something like:

`aws codebuild put-test-report` accepting a payload schema similar to the `reports` stanza in `buildspec.yml` -- `reportARN`, `file-format`, `files` (paths on disk).

**Describe alternatives you've considered**

I could pay for a third-party product (codecov, coveralls) for this functionality, or invest in a self-hosted solution. 

Given that Codebuild already supports coverage reporting natively, it would be nice to not need to build or purchase another tool to see coverage over time.

**Comment by kdaily** [comment]
Hi @blampe,

Thanks for the feature request. I will raise it with the CodeBuild team. You can also request this with the CodeBuild team in the [AWS CodeBuild Forum](https://forums.aws.amazon.com/forum.jspa?forumID=230).

You can currently see the results of test reports over time in CodeBuild (for 30 days) ([reference](https://docs.aws.amazon.com/codebuild/latest/userguide/test-view-reports.html)). Were you thinking of something different?

**Comment by blampe** [comment]
> You can currently see the results of test reports over time in CodeBuild (for 30 days) (reference). Were you thinking of something different?

This is exactly what I was thinking of, the dilemma being that it's only possible to generate these reports when run on AWS (hence this feature request -- to programmatically submit reports from other environments).

**Comment by kdaily** [comment]
Got it, thanks! This would still be something for the CodeBuild team, not the AWS CLI. I'll forward the feature request on to them.

**Comment by tim-finnigan** [comment]
Checking in as this issue hasn't been active in a while. I saw that CodeBuild added this documentation page for creating test reports: https://docs.aws.amazon.com/codebuild/latest/userguide/sample-test-report-cli.html. Does that address the original use case?

If not then we generally recommend reaching out through [AWS Support](https://aws.amazon.com/contact-us/) for feature requests that involve service API functionality. (This is because the APIs are owned by service teams like CodeBuild and used across the AWS CLI/SDKs).

I could also create an internal ticket and forward your request to the CodeBuild team for feedback. Before doing that though, could you summarize the feature request in a sentence or two? Because I see there has been some discussion here and I'm not sure if I fully understand the specific ask.
